lib/src/obs/mod.rs

```diff
@@ -0,0 +1,94 @@
+//! This module defines the operations for object storage using the [object_store](https://docs.rs/object_store/latest/object_store/)
+//! crate. This will enable the user to store objects using local file storage, or cloud storage such as S3 or GCS.
+use crate::err::Error;
+use bytes::Bytes;
+use futures::stream::BoxStream;
+use object_store::local::LocalFileSystem;
+use object_store::parse_url;
+use object_store::path::Path;
+use object_store::ObjectStore;
+use once_cell::sync::Lazy;
+use sha1::{Digest, Sha1};
+use std::env;
+use std::fs;
+use std::sync::Arc;
+use url::Url;
+
+static STORE: Lazy<Arc<dyn ObjectStore>> =
+	Lazy::new(|| match std::env::var("SURREAL_OBJECT_STORE") {
+		Ok(url) => {
+			let url = Url::parse(&url).expect("Expected a valid url for SURREAL_OBJECT_STORE");
+			let (store, _) =
+				parse_url(&url).expect("Expected a valid url for SURREAL_OBJECT_STORE");
+			Arc::new(store)
+		}
+		Err(_) => {
+			let path = env::current_dir().unwrap().join("store");
+			if !path.exists() {
+				fs::create_dir_all(&path)
+					.expect("Unable to create directory structure for SURREAL_OBJECT_STORE");
+			}
```

Comment:
If the path already exists, shouldn't we also make sure that it's a directory?

Reply:
I presumed the `create_dir_all` would handle that! Is that not the case?

Reply:
We are only calling `create_dir_all` when the path doesn't exist. Currently when the path exists, we don't do anything. So we are assuming that if it already exists it's a directory. If a user or some other process creates a file on that path before us, our assumption will be wrong.